mpy-cross link error on macOS

Got the following linker error on macOS building `mpy-cross` with commit `[96aa3a31]`

```
$ make -C mpy-cross
Use make V=1 or set BUILD_VERBOSE in your environment to increase build verbosity.
LINK mpy-cross
Undefined symbols for architecture x86_64:
  "nlr_push_tail", referenced from:
      _nlr_push in nlrx64.o
     (maybe you meant: _nlr_push_tail)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [mpy-cross] Error 1
```

Seems to have fixed the cross-compiler link issue, but building for the STM32F4Discovery board fails.

```
CC ../py/../lib/utils/printf.c
MPY modules/lcd160cr.py
make: *** [build-STM32F4DISC/frozen_mpy/lcd160cr.mpy] Segmentation fault: 11
```

--- a/py/nlrx64.c
+++ b/py/nlrx64.c
@@ -61,6 +61,9 @@ unsigned int nlr_push(nlr_buf_t *nlr) {
     #else
 
     __asm volatile (
+    #if defined(__APPLE__) || defined(__MACH__)
+    "pop    %rbp                \n" // undo function's prelude
+    #endif
     "movq   (%rsp), %rax        \n" // load return %rip
     "movq   %rax, 16(%rdi)      \n" // store %rip into nlr_buf
     "movq   %rbp, 24(%rdi)      \n" // store %rbp into nlr_buf
@@ -70,7 +73,11 @@ unsigned int nlr_push(nlr_buf_t *nlr) {
     "movq   %r13, 56(%rdi)      \n" // store %r13 into nlr_buf
     "movq   %r14, 64(%rdi)      \n" // store %r14 into nlr_buf
     "movq   %r15, 72(%rdi)      \n" // store %r15 into nlr_buf
+    #if defined(__APPLE__) || defined(__MACH__)
+    "jmp    _nlr_push_tail      \n" // do the rest in C
+    #else
     "jmp    nlr_push_tail       \n" // do the rest in C
+    #endif
     );
 
     #endif

